solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -589,23 +622,28 @@ contract RandomBeacon is Ownable {
             .submitEntry(submitterIndex, entry, group);
 
         if (inactiveMembers.length > 0) {
-            // TODO: Make the duration a governable parameter.
-            punishOperators(inactiveMembers, 2 weeks);
+            banFromRewards(inactiveMembers, sortitionPoolRewardsBanDuration);
         }
 
         if (slashingAmount > 0) {
             address[] memory groupMembers = sortitionPool.getIDOperators(
                 group.members
             );
 
-            // slither-disable-next-line reentrancy-events
-            emit RelayEntryDelaySlashed(
-                currentRequestId,
-                slashingAmount,
-                groupMembers
-            );
-
-            staking.slash(slashingAmount, groupMembers);
+            try staking.slash(slashingAmount, groupMembers) {
+                // slither-disable-next-line reentrancy-events
+                emit RelayEntryDelaySlashed(
+                    currentRequestId,
+                    slashingAmount,
+                    groupMembers
+                );
+            } catch {
+                emit RelayEntryDelaySlashingFailed(
```

Comment:
Can we add a comment explaining this try-catch? Say:
```
// Should never happen but we want to ensure a non-critical path
// failure from an external contract does not stop group members
// from submitting a valid relay entry.
```

Here, and at the line 712, before we emit `BanRewardsFailed`.

Reply:
See https://github.com/keep-network/keep-core/pull/2707/commits/6be521c1ffbd85c473e05aee899ab786382f4eba